packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -723,40 +727,61 @@ class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin
       mouseCursor: effectiveMouseCursor,
       child: CompositedTransformTarget(
         link: _layerLink,
-        child: _RangeSliderRenderObjectWidget(
-          values: _unlerpRangeValues(widget.values),
-          divisions: widget.divisions,
-          labels: widget.labels,
-          sliderTheme: sliderTheme,
-          textScaleFactor: effectiveTextScale,
-          screenSize: screenSize(),
-          onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
-          onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
-          onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
-          state: this,
-          semanticFormatterCallback: widget.semanticFormatterCallback,
-          hovering: _hovering,
+        child: OverlayPortal(
+          controller: _valueIndicatorOverlayPortalController,
+          overlayChildBuilder: (BuildContext context) {
+            return _buildValueIndicator(sliderTheme.showValueIndicator);
+          },
+          child: _RangeSliderRenderObjectWidget(
+            values: _unlerpRangeValues(widget.values),
+            divisions: widget.divisions,
+            labels: widget.labels,
+            sliderTheme: sliderTheme,
+            textScaleFactor: effectiveTextScale,
+            screenSize: screenSize(),
+            onChanged: _enabled && (widget.max > widget.min) ? _handleChanged : null,
+            onChangeStart: widget.onChangeStart != null ? _handleDragStart : null,
+            onChangeEnd: widget.onChangeEnd != null ? _handleDragEnd : null,
+            state: this,
+            semanticFormatterCallback: widget.semanticFormatterCallback,
+            hovering: _hovering,
+          ),
         ),
       ),
     );
   }
 
   final LayerLink _layerLink = LayerLink();
-
-  OverlayEntry? overlayEntry;
+  Widget _buildValueIndicator(ShowValueIndicator? showValueIndicator) {
+    late final Widget valueIndicator = CompositedTransformFollower(
+      link: _layerLink,
+      child: _ValueIndicatorRenderObjectWidget(state: this),
+    );
+    late final Widget valueIndicatorWhenDragged = ValueListenableBuilder<bool>(
+      valueListenable: _showValueIndicatorForDragged,
+      builder: (BuildContext context, bool showValueIndicator, Widget? child) {
+        return showValueIndicator ? valueIndicator : const SizedBox.shrink();
+      },
+    );
+    return switch (showValueIndicator) {
+      ShowValueIndicator.alwaysVisible => valueIndicator,
+      ShowValueIndicator.onlyForDiscrete =>
+        widget.divisions != null ? valueIndicatorWhenDragged : const SizedBox.shrink(),
+      ShowValueIndicator.onlyForContinuous =>
+        widget.divisions == null ? valueIndicatorWhenDragged : const SizedBox.shrink(),
+      ShowValueIndicator.onDrag || ShowValueIndicator.always => valueIndicatorWhenDragged,
+      _ => const SizedBox.shrink(),
```

Comment:
Consider write out the enum case here for readability / enabling exhaustiveness check?